software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+-- [tagDataList] 5,30,200,395
+-- [tagLabel] 205,30,395,51
+-- [editData] 205,56,395,395
+-- [loadImgButt] 219,41
+-- [clearImgButt] 256,41
+-- [expandImgButt] 300,41
+-- [setCaptureButt] 340,41
+-- [snapCaptureButt] 372,41
+local sCaptureAreaGeometry
+-- [captureAreaRect] 400,400,800,800 (width and height)
+-- square icons set to middle of height and equally spaced across width, 1/5th each
+
+on setupAsOf3FEB2016
+   set the defaultstack to "HowstrBeta"
+   -- setup the window
+   set the decorations of stack "howstrBeta" to "title"
+   --   set the decorations of this stack to empty
+   --   set the windowShape of this stack to 2868
+   set the windowshape of stack "howstrBeta" to 0
+   set the minHeight of stack "howstrBeta" to 400
+   set the minWidth of stack "howstrBeta" to 500
+   --   -- shouldn't be necessary forever, but a good hack for now
+   set the foregroundColor of card 1 of stack "Ask Dialog" to "0,220,220"
+   --   set the opaque  of button "Ok" of card 1 of stack "Ask Dialog" to false
+   --   set the opaque  of button "Cancel" of card 1 of stack "Ask Dialog" to false
+   set the foregroundColor of card 1 of stack "Answer Dialog" to "0,220,220"
+   --   set the opaque  of button "Ok" of card 1 of stack "Answer Dialog" to false
+   --   set the opaque  of button "close current" of card 1 of stack "Answer Dialog" to false
+   --   set the opaque  of button "close" of card 1 of stack "Answer Dialog" to false
+   
+   repeat with i = 1 to the number of buttons of card 1 of stack "Ask Dialog"
+      --      put the name of button i of group "mygroup" after field 1
+      set the opaque  of button i of card 1 of stack "Ask Dialog" to false
+   end repeat
+   repeat with i = 1 to the number of buttons of card 1 of stack "Answer Dialog"
+      --      put the name of button i of group "mygroup" after field 1
+      set the opaque  of button i of card 1 of stack "Answer Dialog" to false
+   end repeat
+   
+   put empty into sMain
+   put empty into sNowDoc
+   put empty into sNowItem
+   put empty into sPointerTags
+   -- initiate development logging
+   put "initiate" into xX["write"];    devLog xX
+   quickModeSetup sMain,sNowDoc,sNowItem,sPointerTags -- quickiator
+   put empty into sMainShort
+   adjList sMain,sMainShort
+   put empty into sGraphData
+   tempConvertData sMainShort,sGraphData,sNowDoc -- backend alpha and frontend alpha used different data structures
+   put empty into sGraphDrawing
+   --   put 8 into sNowFocus 
+   --   -- find a good node to focus on; highest is probably the last place they worked
+   --   put the keys of sMainShort[sNowDoc] into tKeys
+   --   sort tKeys numeric descending
+   --   repeat for each line tAkey in tKeys
+   --      if sMainShort[sNowDoc][tAkey]["type"] is "state" then
+   --         put tAkey into sNowFocus
+   --         exit repeat
+   --      end if
+   --   end repeat
+   --   -- now middleOut can start
+   middleOut sGraphData,sGraphDrawing,sNowFocus -- displayerang.livecode
```

Comment:
@matthewmaier I think it would be better if control of the graph were handled in a single module (as I mentioned before perhaps a group behavior script). I would be inclined to pass it the data and have that trigger redrawing `middleOut`.